### PR TITLE
feat: improves Lambda data flushing and extension signaling

### PR DESCRIPTION
**Pull request by trentm**
Lambda instrumentation will now signal the Elastic Lambda extension
when a Lambda invocation is done (via a '?flushed=true' empty intake
request). This should allow for better Lambda function latency as the
ext no longer needs to wait for Logs API platform.runtimeDone.

As well, the time that `Agent.flush()` at the end of a Lambda invocation
will wait for inflight events has been reduced from 1s to 100ms to also
attempt to reduce latency.

Closes: #2485
Fixes: #2166

### Checklist

- [x] Cannot merge this change to main until https://github.com/elastic/apm-aws-lambda/pull/132 (which includes some ext crash fixes this crash: https://github.com/elastic/apm-aws-lambda/issues/133) and there is a release of the extension.
- [x] Update to v11.0.0 of the [http client](https://github.com/elastic/apm-nodejs-http-client) when https://github.com/elastic/apm-nodejs-http-client/pull/178 is merged and published.
- [x] Note somehow that new agent support should only be used with `>v0.0.3` of the extension. The current v0.0.3 of the extension is crashy (https://github.com/elastic/apm-aws-lambda/issues/133) with the `?flushed=true` signaling.
- [x] Implement code
- [x] Add tests
- [x] Add CHANGELOG.asciidoc entry

### CODE REVIEW

**Review by astorm — APPROVED** [review]
Does what it says on the box.  Approving, with the already made "this relies on v11 of the client to be released before we release".

**Comment by trentm** [comment]
elastic-apm-http-client@11.0.0 is published

File: lib/agent.js [hunk 1]
```diff
@@ -621,15 +642,11 @@ Agent.prototype.flush = function (cb) {
...
-  // This 1s timeout is a subjective balance between "long enough for spans
-  // and errors to reasonably encode" and "short enough to not block data
-  // being reported to APM server".
-  const INS_FLUSH_TIMEOUT_MS = 1000
   const boundCb = cb && this._instrumentation.bindFunction(cb)
 
   // If there are no inflight events then avoid creating additional objects.
   if (this._inflightEvents.size === 0) {
-    this._transport.flush(boundCb)
+    this._transport.flush({ lambdaEnd }, boundCb)
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/agent.js:649]: Done in commit cec204b

File: package.json [hunk 2]
```diff
@@ -86,7 +88,7 @@
     "basic-auth": "^2.0.1",
     "cookie": "^0.4.0",
     "core-util-is": "^1.0.2",
-    "elastic-apm-http-client": "^10.4.0",
+    "elastic-apm-http-client": "elastic/apm-nodejs-http-client#trentm/lambda-coord",
```
[the remark this answers is not in the document]
Comment (trentm) [reply package.json]: Done in commit cec204b

**Pull request merged**